app/helpers/react_components/student/mentoring_session/mentor.rb

```diff
@@ -0,0 +1,53 @@
+module ReactComponents
+  module Student
+    class MentoringSession
+      class Mentor
+        class << self
+          attr_writer :mentor_keys, :relationship_keys
+
+          def mentor_keys
+            @mentor_keys || %i[name handle bio languages_spoken avatar_url formatted_reputation pronouns]
+          end
+
+          def relationship_keys
+            @relationship_keys || %i[num_discussions]
+          end
+        end
+
+        def initialize(mentor, relationship)
+          @mentor = mentor
+          @relationship = relationship || NullRelationship.new
+        end
+
+        def as_json
+          return nil unless mentor
+
+          mentor_json.merge(relationship_json)
+        end
+
+        private
+        attr_reader :mentor, :relationship
+
+        def mentor_json
+          json(mentor, self.class.mentor_keys)
+        end
+
+        def relationship_json
+          json(relationship, self.class.relationship_keys)
+        end
+
+        def json(object, keys)
+          keys.index_with do |key|
+            object.send(key)
+          end
+        end
+      end
+
+      class NullRelationship
+        def num_discussions
+          0
+        end
+      end
+    end
+  end
+end
```

Comment:
@kntsoriano I like your instinct here to refactor this to be cleaner. I've made a few changes though:
1. I've inlined it into the class. I don't internal helper components (such as this) to be kept internal to the place they're used. I've mentioned this on another PR as well.
2. I've changed it to be an internal serializer. Let's stick with the same patterns we use elsewhere rather than introducing new ones.
3. I've removed the stubbing of the attributes. I get why you've done this to make the tests faster, but in doing this we don't actually check anywhere that the right data is being passed in (ie we're not actually checking the correct keys). Removing all this and changing the one test that used them to check the keys simplified the code lots and also meant the test is much more robust in ensuring the correct data is collected, which is the most important thing in this test.
4. Renamed the bits called `json` to `hash` as they weren't actually doing anything with JSON.

Reply:
1. I think this is why I put it within the `MentoringSession` namespace as this would imply that it is internal to the component that it is used with.
2. Okay!
3. We check in the system test that the correct attributes are used.
4. Okay!

Reply:
Two thoughts on 3:
1. We check some of them in the system test, but not all of them. That might be a smell (in a good way), because it might mean we should delete some from the serializer that aren't used. But I feel uncomfortable that we're potentially relying on data downstream but not checking for it here. 
2. I really like unit tests! 🙂 So although we might check things in a system test, that for me is a test of last resort - it's our final safety net. I think it's really important that the individual components within Rails get tested at that level too.

Reply:
1. We check all of them in the system tests, and that should be the case because we want the user to see all of these information on their screen.
2. I use system tests as a way to discover how the system will go as there is an overarching goal to get to. It's not really a last resort for me as I think we should test most primarily how the user will use the system and unit tests should test how the units interact with each other. My fear with having unit tests and system tests testing the same thing is that we will be changing multiple tests for a single change. This is one of the frustrations you've raised and I want to avoid this.